cunumeric/runtime.py

```diff
@@ -241,45 +256,59 @@ def destroy(self) -> None:
             self._report_coverage()
         self.destroyed = True
 
-    def create_scalar(self, array: memoryview, dtype, shape=None, wrap=False):
+    def create_scalar(
+        self,
+        array: memoryview,
+        dtype: np.dtype[Any],
+        shape: Optional[NdShape] = None,
+    ) -> Future:
         data = array.tobytes()
         buf = struct.pack(f"{len(data)}s", data)
-        future = self.legate_runtime.create_future(buf, len(buf))
-        if wrap:
-            assert all(extent == 1 for extent in shape)
-            assert shape is not None
-            store = self.legate_context.create_store(
-                dtype,
-                shape=shape,
-                storage=future,
-                optimize_scalar=True,
-            )
-            result = DeferredArray(self, store, dtype=dtype)
-        else:
-            result = future
-        return result
+        return self.legate_runtime.create_future(buf, len(buf))
+
+    def create_wrapped_scalar(
+        self, array: memoryview, dtype: np.dtype[Any], shape: NdShape
+    ) -> DeferredArray:
+        future = self.create_scalar(array, dtype, shape)
+        assert all(extent == 1 for extent in shape)
+        store = self.legate_context.create_store(
+            dtype,
+            shape=shape,
+            storage=future,
+            optimize_scalar=True,
+        )
+        return DeferredArray(self, store, dtype=dtype)
 
     def set_next_random_epoch(self, epoch: int) -> None:
         self.current_random_epoch = epoch
 
-    def get_next_random_epoch(self):
+    def get_next_random_epoch(self) -> int:
         result = self.current_random_epoch
         # self.current_random_epoch += 1
         return result
 
-    def is_point_type(self, dtype):
-        if len(dtype) == 6 and dtype[0:5] == "Point":
+    def is_point_type(self, dtype: Union[str, np.dtype[Any]]) -> bool:
+        if (
+            isinstance(dtype, str)
+            and len(dtype) == 6
```

Comment:
I guess technically `len` of a real dtype works (in some way) but adding the str check seems safer/clearer about things.

Reply:
we will create proper numpy dtypes for the point types (#385), so the `dtype`'s type will become `np.dtype[Any]` eventually.